tools/migration/internal/logger.go

```diff
@@ -0,0 +1,46 @@
+package internal
+
+import (
+	"io"
+	"log"
+	"os"
+)
+
+// Logger logs migration events to disk and, if initialized with verbose,
+// stdout too.
+type Logger struct {
+	logFile *os.File
+}
+
+// NewLogger creates a new Logger.  All log writes go to f, the logging file.
+// If the verbose flag is true all log writes also go to stdout.
+func NewLogger(f *os.File, verbose bool) *Logger {
```

Comment:
I suggest relaxing `os.File` to `io.Writer` here, since no other behaviour of File is needed. This could also simplify testing, where you could point this to a bytestream.

Reply:
I'm going to make it a `WriteCloser` so I can still close the file from within the logger

Reply:
(without casting)